File: frontend/src/api/modules/traffic.ts
import http from '@/api/http'

export interface TrafficConfig {
  id: number
  interfaceName: string
  monthlyLimit: number
  resetDay: number
  enabled: boolean
}

export interface TrafficConfigCreate {
  interfaceName: string
  monthlyLimit: number
  resetDay: number
  enabled: boolean
}

export interface InterfaceInfo {
  name: string
  ipv4: string[]
  ipv6: string[]
  mac: string
  status: string
}

export interface TrafficStatsRequest {
  interfaceName: string
  startTime: string
  endTime: string
  groupBy: 'hour' | 'day'
}

export interface TrafficStatsItem {
  timestamp: string
  bytesSent: number
  bytesRecv: number
}

export interface TrafficStatsResponse {
  interfaceName: string
  items: TrafficStatsItem[]
  totalSent: number
  totalRecv: number
}

export interface TrafficSummaryItem {
  interfaceName: string
  monthlyLimit: number
  resetDay: number
  periodStart: string
  periodEnd: string
  totalSent: number
  totalRecv: number
  totalUsed: number
  usedPercent: number
  enabled: boolean
}

export const trafficApi = {
  listConfigs: () => http.get<{ data: TrafficConfig[] }>('/traffic/configs'),
  createConfig: (params: TrafficConfigCreate) => http.post('/traffic/configs', params),
  deleteConfig: (interfaceName: string) => http.post('/traffic/configs/del', { interfaceName }),
  listInterfaces: () => http.get<{ data: InterfaceInfo[] }>('/traffic/interfaces'),
  getStats: (params: TrafficStatsRequest) => http.post<{ data: TrafficStatsResponse }>('/traffic/stats', params),
  getSummary: () => http.get<{ data: TrafficSummaryItem[] }>('/traffic/summary'),
}

File: backend/app/dto/traffic.go
package dto

import "time"

// TrafficConfigCreate 创建/更新流量监控配置
type TrafficConfigCreate struct {
	InterfaceName string `json:"interfaceName" binding:"required"`
	MonthlyLimit  uint64 `json:"monthlyLimit"`
	ResetDay      int    `json:"resetDay" binding:"required,min=1,max=28"`
	Enabled       bool   `json:"enabled"`
}

// TrafficConfigDTO 流量监控配置响应
type TrafficConfigDTO struct {
	ID            uint   `json:"id"`
	InterfaceName string `json:"interfaceName"`
	MonthlyLimit  uint64 `json:"monthlyLimit"`
	ResetDay      int    `json:"resetDay"`
	Enabled       bool   `json:"enabled"`
}

// TrafficStatsRequest 流量统计查询请求
type TrafficStatsRequest struct {
	InterfaceName string `json:"interfaceName" binding:"required"`
	StartTime     string `json:"startTime" binding:"required"` // RFC3339 or 2006-01-02
	EndTime       string `json:"endTime" binding:"required"`
	GroupBy       string `json:"groupBy"` // "hour" or "day", default "day"
}

// TrafficStatsItem 流量统计数据点
type TrafficStatsItem struct {
	Timestamp string `json:"timestamp"`
	BytesSent uint64 `json:"bytesSent"`
	BytesRecv uint64 `json:"bytesRecv"`
}

// TrafficStatsResponse 流量统计查询响应
type TrafficStatsResponse struct {
	InterfaceName string             `json:"interfaceName"`
	Items         []TrafficStatsItem `json:"items"`
	TotalSent     uint64             `json:"totalSent"`
	TotalRecv     uint64             `json:"totalRecv"`
}

// TrafficSummaryItem 单网卡的当前计费周期汇总
type TrafficSummaryItem struct {
	InterfaceName string    `json:"interfaceName"`
	MonthlyLimit  uint64    `json:"monthlyLimit"`
	ResetDay      int       `json:"resetDay"`
	PeriodStart   time.Time `json:"periodStart"`
	PeriodEnd     time.Time `json:"periodEnd"`
	TotalSent     uint64    `json:"totalSent"`
	TotalRecv     uint64    `json:"totalRecv"`
	TotalUsed     uint64    `json:"totalUsed"`
	UsedPercent   float64   `json:"usedPercent"`
	Enabled       bool      `json:"enabled"`
}

// TrafficDeleteConfig 删除流量监控配置
type TrafficDeleteConfig struct {
	InterfaceName string `json:"interfaceName" binding:"required"`
}
